app/src/main/java/io/github/easyalbum/MainActivity.kt

```diff
@@ -55,11 +60,15 @@ class MainActivity : AppCompatActivity() {
 
     private class TestMediaFilter(private val opt: Option) : MediaFilter {
         override fun accept(media: MediaData): Boolean {
-            return when (opt) {
+            val mediaType = when (opt) {
                 Option.VIDEO -> media.isVideo
                 Option.IMAGE -> !media.isVideo
                 else -> true
             }
+            val valid = ((media.fileSize in ((minSize + 1) until maxSize))
+                    && (media.width > minWidth)
+                    && (media.height > minHeight))
+            return if (opt == Option.ALL) valid else mediaType && valid
```

Comment:
"opt == Option.ALL" is unnessary, because the value 'opt' has only three cases : "IMAGE, VIDEO, ALL"